skorch/tests/test_classifier.py

```diff
@@ -208,3 +208,27 @@ def test_target_2d_raises(self, net, data):
 
         assert exc.value.args[0] == (
             "The target data should be 1-dimensional.")
+
+    def test_custom_loss_does_not_call_sigmoid(
+            self, net_cls, data, module_cls, monkeypatch):
+        mock = Mock(side_effect=lambda x: x)
+        monkeypatch.setattr(torch.nn.functional, "sigmoid", mock)
+
+        net = net_cls(module_cls, max_epochs=20, lr=0.1, criterion=nn.MSELoss)
+        X, y = data
+        net.fit(X, y)
+
+        net.predict_proba(X)
+        mock.assert_not_called()
```

Comment:
Minor pet peeve of mine: I don't like methods on mock such as `assert_not_called`. The reason is that if there is a typo (e.g. `mock.asert_not_called()`), the test just passes. For this particular method, I like `mock.call_count == 1` better for the given reason.